sympy/solvers/ode/tests/test_ode.py

```diff
@@ -33,6 +33,17 @@
 # dsolve() to return different results on different machines)
 
 
+def _ode_solver_test(ode_examples,our_hint='default'):
+    for example in ode_examples:
+        eq = ode_examples[example]['eq']
+        sol = ode_examples[example]['sol']
+        assert our_hint in classify_ode(eq)
+        sols = [sol,constant_renumber(sol)]
+        sols += [sols[-1].expand()]
+        assert dsolve(eq,hint=our_hint).rhs in sols
+        assert checkodesol(eq, sol)[0]
```

Comment:
Using `assert checkodesol` liks this does not work:
```
In [20]: eq = Eq(f(x).diff(x), 0)                                                                        

In [21]: sol = Eq(f(x), x)                                                                               

In [22]: checkodesol(eq, sol)                                                                            
Out[22]: (False, 1)

In [23]: bool(checkodesol(eq, sol))                                                                      
Out[23]: True

In [24]: assert checkodesol(eq, sol)  # doesn't raise  
```
This should assert exactly what the output of `checkodesol` should be.

Reply:
You should arrange it so that the exception that is raised from a failing test has a nice message showing the name of the failed example and explaining why it failed.

Reply:
so what should replace `assert checkodesol(eq, sol)[0]`?

Reply:
@oscarbenjamin I think `assert checkodesol(eq, sol)[0]` will always work because `checkodesol(eq, sol)` always returns `(boolean,residue)`

Reply:
No it doesn't:
```
In [1]: eq = f(x).diff(x)**2 - 1                                                                                                  

In [2]: sol = dsolve(eq)                                                                                                          

In [3]: sol                                                                                                                       
Out[3]: [f(x) = C₁ + x, f(x) = C₁ - x]

In [4]: checkodesol(eq, sol)                                                                                                      
Out[4]: [(True, 0), (True, 0)]
```
This test should assert precisely what the output of checkodesol should be like:
```
assert checkodeol(...) == (True, 0)
```
except that the expected return value depends on the number of solutions

Reply:
It is generally not good to use `assert f()` because most objects in Python are truthy.